src/ompl/tools/thunder/src/SPARSdb.cpp

```diff
@@ -1682,7 +1680,12 @@ void ompl::geometric::SPARSdb::connectGuards(Vertex v, Vertex vp)
     Edge e = (boost::add_edge(v, vp, g_)).first;
 
     // Add associated properties to the edge
-    edgeWeightProperty_[e] = costFunction(v, vp);  // TODO: use this value with astar
+    if (edge_weight.has_value()) {
+        edgeWeightProperty_[e] = edge_weight.value().value();
```

Comment:
Can the second value() be null here?

Reply:
No I dont think so, the constructor of the ob::Cost class sets the value to zero by default